hypothesis-python/src/hypothesis/core.py

```diff
@@ -962,8 +962,8 @@ def fuzz_one_input(
 
 
 def given(
-    *_given_arguments: Union[SearchStrategy, InferType],
-    **_given_kwargs: Union[SearchStrategy, InferType],
+    *_given_arguments: Union[SearchStrategy[Ex], InferType],
+    **_given_kwargs: Union[SearchStrategy[Ex], InferType],
```

Comment:
```
 - error: Type of "given" is partially unknown
    Type of "given" is "(*_given_arguments: SearchStrategy[Unknown] | InferType, **_given_kwargs: SearchStrategy[Unknown] | InferType) -> (((...) -> Coroutine[Any, Any, None] | None) -> ((...) -> None))" (reportUnknownMemberType)
```

Reply:
Weird! This happens when using `@hypothesis.given`, but not when using `@given`

Reply:
That's very weird. Perhaps a bug in pyright

Reply:
https://github.com/microsoft/pyright/issues/3334